src/SeleniumLibrary/locators/elementfinder.py

```diff
@@ -220,6 +221,12 @@ def _find_by_tag_name(self, criteria, tag, constraints, parent):
             parent.find_elements(By.TAG_NAME, criteria), tag, constraints
         )
 
+    def _find_by_data_locator(self, criteria, tag, constraints, parent):
+        name, value = criteria.split(":", 2)
+        local_criteria = f'//*[@data-{name}="{value}"]'
+        return self._find_by_xpath(local_criteria, tag, constraints, parent)
+
+
```

Comment:
`name, value = criteria.split(":", 2)` fails with somewhat cryptic ValueError if `criteria` has no `:` in it. Should probably check that (or catch ValueError) and provide a better error to users.

Also unnecessary empty line.